STM32WL - LoRa sleep-mode

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

Were trying to reduce sleep current like the SX126X driver by using the following
"stm32wl-lora-driver.sleep-mode": 1

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

However it doesn't even compile, fixing the obvious errors results in the radio not working properly and failing to join a network.

```
Compile [ 13.0%]: STM32WL_LoRaRadio.cpp
[Warning] STM32WL_LoRaRadio.cpp@218,23: 'sleep_for' is deprecated: Pass a chrono duration, not an integer millisecond count. For example use `5s` rather than `5000`. [since mbed-os-6.0.0] [-Wdeprecated-declarations]
[Warning] STM32WL_LoRaRadio.cpp@224,25: 'read_ms' is deprecated: Use the Chrono-based elapsed_time method.  If integer milliseconds are needed, you can use `duration_cast<milliseconds>(elapsed_time()).count()` [since mbed-os-6.0.0] [-Wdeprecated-declarations]
[Error] STM32WL_LoRaRadio.cpp@720,9: use of undeclared identifier 'wait_us'
[Error] STM32WL_LoRaRadio.cpp@735,5: use of undeclared identifier 'sleep_state'
[ERROR] .\mbed-os\connectivity\drivers\lora\TARGET_STM32WL\STM32WL_LoRaRadio.cpp:218:23: warning: 'sleep_for' is deprecated: Pass a chrono duration, not an integer millisecond count. For example use `5s` rather than `5000`. [since mbed-os-6.0.0] [-Wdeprecated-declarations]
    rtos::ThisThread::sleep_for(sleep_duration);
                      ^
./mbed-os/rtos/include/rtos\ThisThread.h:215:1: note: 'sleep_for' has been explicitly marked deprecated here
MBED_DEPRECATED_SINCE("mbed-os-6.0.0", "Pass a chrono duration, not an integer millisecond count. For example use `5s` rather than `5000`.")
^
./mbed-os/platform/include\platform/mbed_toolchain.h:429:37: note: expanded from macro 'MBED_DEPRECATED_SINCE'
#define MBED_DEPRECATED_SINCE(D, M) MBED_DEPRECATED(M " [since " D "]")
                                    ^
./mbed-os/platform/include\platform/mbed_toolchain.h:410:43: note: expanded from macro 'MBED_DEPRECATED'
#define MBED_DEPRECATED(M) __attribute__((deprecated(M)))
                                          ^
.\mbed-os\connectivity\drivers\lora\TARGET_STM32WL\STM32WL_LoRaRadio.cpp:224:25: warning: 'read_ms' is deprecated: Use the Chrono-based elapsed_time method.  If integer milliseconds are needed, you can use `duration_cast<milliseconds>(elapsed_time()).count()` [since mbed-os-6.0.0] [-Wdeprecated-declarations]
    while (elapsed_time.read_ms() < (int) max_carrier_sense_time) {
                        ^
./mbed-os/drivers/include/drivers\Timer.h:85:5: note: 'read_ms' has been explicitly marked deprecated here
    MBED_DEPRECATED_SINCE("mbed-os-6.0.0", "Use the Chrono-based elapsed_time method.  If integer milliseconds are needed, you can use `duration_cast<milliseconds>(elapsed_time()).count()`")
    ^
./mbed-os/platform/include\platform/mbed_toolchain.h:429:37: note: expanded from macro 'MBED_DEPRECATED_SINCE'
#define MBED_DEPRECATED_SINCE(D, M) MBED_DEPRECATED(M " [since " D "]")
                                    ^
./mbed-os/platform/include\platform/mbed_toolchain.h:410:43: note: expanded from macro 'MBED_DEPRECATED'
#define MBED_DEPRECATED(M) __attribute__((deprecated(M)))
                                          ^
.\mbed-os\connectivity\drivers\lora\TARGET_STM32WL\STM32WL_LoRaRadio.cpp:720:9: error: use of undeclared identifier 'wait_us'
        wait_us(3500);
        ^
.\mbed-os\connectivity\drivers\lora\TARGET_STM32WL\STM32WL_LoRaRadio.cpp:735:5: error: use of undeclared identifier 'sleep_state'
    sleep_state = 0x00;
    ^
2 warnings and 2 errors generated.
```

#### Target(s) affected by this defect ?
NUCLEO_WL55JC

#### Toolchain(s) (name and version) displaying this defect ?
ARMC6

#### What version of Mbed-os are you using (tag or sha) ?
Master SHA-1: 91b793c4fd1ad7f02b8db5a2e0d77daf63e5c836

<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->


#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
mbed-cli --version
1.10.5
#### How is this defect reproduced ? 
Add "stm32wl-lora-driver.sleep-mode": 1 to mbed_app

<!--
    For connectivity issues it is recommended to enable mbed trace and attach the traces 
    generated by the logger to the issue. 
-->

--- a/connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp
+++ b/connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp
@@ -38,6 +38,7 @@ SPDX-License-Identifier: BSD-3-Clause
 #include "ThisThread.h"
 #include "Timer.h"
 #include "STM32WL_LoRaRadio.h"
+#include "mbed_wait_api.h"
 
 #ifndef DEBUG_STDIO
 #define DEBUG_STDIO 0
@@ -728,18 +729,18 @@ void STM32WL_LoRaRadio::wakeup()
 void STM32WL_LoRaRadio::sleep(void)
 {
     DEBUG_PRINTF("STM32WL_LoRaRadio::sleep\n");
-#if MBED_CONF_STM32WL_LORA_DRIVER_SLEEP_MODE == 1
-    // cold start, power consumption 160 nA
-    sleep_state = 0x00;
-#endif
 
     /* switch the antenna OFF by SW */
     set_antenna_switch(RBI_SWITCH_OFF);
     Radio_SMPS_Set(SMPS_DRIVE_SETTING_DEFAULT);
 
-
+#if MBED_CONF_STM32WL_LORA_DRIVER_SLEEP_MODE == 1
+    // cold start, power consumption 160 nA
+    uint8_t sleep_state = 0x00;
+#else
     // warm start set , power consumption 600 nA
     uint8_t sleep_state = 0x04;
+#endif
     write_opmode_command(RADIO_SET_SLEEP, &sleep_state, 1);
 
     _operating_mode = MODE_SLEEP;

